component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,561 @@
+     * @return true or false
+     * @throws VoiceOTPException
+     */
+    public static boolean isVoiceOTPDisableForLocalUser(String username, AuthenticationContext context)
+            throws VoiceOTPException {
+
+        UserRealm userRealm;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = realmService.getTenantUserRealm(tenantId);
+            username = MultitenantUtils.getTenantAwareUsername(String.valueOf(username));
+            boolean isEnablingControlledByUser = isVoiceOTPEnableOrDisableByUser(context);
+            if (userRealm != null) {
+                if (isEnablingControlledByUser) {
+                    Map<String, String> claimValues = userRealm.getUserStoreManager().getUserClaimValues(username,
+                            new String[]{VoiceOTPConstants.USER_VOICEOTP_DISABLED_CLAIM_URI}, null);
+                    return Boolean.parseBoolean(claimValues.get(VoiceOTPConstants.USER_VOICEOTP_DISABLED_CLAIM_URI));
+                }
+            } else {
+                throw new VoiceOTPException("Cannot find the user realm for the given tenant domain : " + CarbonContext
+                        .getThreadLocalCarbonContext().getTenantDomain());
+            }
+        } catch (UserStoreException e) {
+            throw new VoiceOTPException("Failed while trying to access userRealm of the user : " + username, e);
+        }
+        return false;
+    }
+
+    /**
+     * Update the mobile number (user attribute) in user's profile.
+     *
+     * @param username  the Username
+     * @param attribute the Attribute
+     * @throws VoiceOTPException
+     * @throws UserStoreException
+     */
+    public static void updateUserAttribute(String username, Map<String, String> attribute, String tenantDomain)
+            throws VoiceOTPException, UserStoreException {
+
+        try {
+            // updating user attributes is independent from tenant association.not tenant association check needed here.
+            UserRealm userRealm;
+            // user is always in the super tenant.
+            userRealm = VoiceOTPUtils.getUserRealm(tenantDomain);
+            if (userRealm == null) {
+                throw new VoiceOTPException("The specified tenant domain " + tenantDomain + " does not exist.");
+            }
+            // check whether user already exists in the system.
+            VoiceOTPUtils.verifyUserExists(username, tenantDomain);
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            userStoreManager.setUserClaimValues(username, attribute, null);
+        } catch (AuthenticationFailedException e) {
+            throw new VoiceOTPException("Exception occurred while connecting " +
+                    "to User Store: Authentication is failed. ", e);
```

Comment:
Userstore name is not mentioned. If it is not required, change the error message accordingly.